sourced/ml/algorithms/token_parser.py

```diff
@@ -166,6 +169,20 @@ def ret(name):
             if last:
                 yield from ret(last)
 
+    def split(self, token):
+        """Splits a single identifier
```

Comment:
we do not use such style.
```
        """
        Splits a single identifier
        """
```
or 
```
        """Splits a single identifier"""
```

And please add `.` in the end.

Please check all docstrings